src/MongoDB.Driver/MongoClient.cs

```diff
@@ -33,7 +33,7 @@
 namespace MongoDB.Driver
 {
     /// <inheritdoc/>
-    public class MongoClient : MongoClientBase
+    public class MongoClient : MongoClientBase, IDisposable
```

Comment:
You probably will remove `IDisposable`, and FYI the compiler warning I think you got can be suppressed using:

        [SuppressMessage("Microsoft.Design", "CA1001:TypesThatOwnDisposableFieldsShouldBeDisposable")] // ignore that LibMongoCryptControler is IDisposable